Source/Redemption/Private/PowerUpComponent.cpp

```diff
@@ -83,4 +87,24 @@ void UPowerUpComponent::ChangeActivePowerUp(EActivePowerUp NewPower)
 {
 	CurrentPowerUp = NewPower;
 	ChangeMappingContext();
+}
+
+//will spawn the bone
+void UPowerUpComponent::ExecuteBonePowerUp()
+{
+	if (BoneBP)
+	{
+		if (ACharacter* OwnerChar = Cast<ACharacter>(GetOwner()))
+		{
+			FVector FowardVec = OwnerChar->GetActorForwardVector();
+			FVector SpawnPos = OwnerChar->GetActorLocation() + FowardVec * 100;
+
+			FRotator CharacterRotation = FowardVec.Rotation();
+
+			FVector BoneDirection = FowardVec + FVector(0.0, 0.0, 0.3);
+
+			ABone* Bone = GetWorld()->SpawnActor<ABone>(BoneBP, SpawnPos, FRotator::ZeroRotator);
+			Bone->GetMesh()->AddImpulse(BoneDirection * BONE_SPEED);
+		}
+	}
 }
```

Comment:
Can we add something to delete the bone from the world after some amount of time for optimization purposes?

Reply:
it will be deleted after it hits something

Reply:
if you want I can put a timer to delete it even if it does not hit anything